src/mlpack/tests/main_tests/random_forest_test.cpp

```diff
@@ -186,6 +186,26 @@ BOOST_AUTO_TEST_CASE(RandomForestMinimumLeafSizeTest)
   Log::Fatal.ignoreInput = false;
 }
 
+/**
+ * Make sure maximum depth specified is always a positive number.
+ */
+BOOST_AUTO_TEST_CASE(RandomForestMaximumDepthTest)
+{
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    BOOST_FAIL("Cannot load train dataset vc2.csv!");
+
+  arma::Row<size_t> labels;
+  if (!data::Load("vc2_labels.txt", labels))
+    BOOST_FAIL("Cannot load labels for vc2_labels.txt");
+
+  SetInputParam("maximum_depth", (int) 0); // Invalid.
```

Comment:
I think 0 should be valid---that should mean there is no limit on the depth.

Reply:
Updated :)